src/org/labkey/test/util/TestDataGenerator.java

```diff
@@ -411,4 +413,22 @@ public static TestDataGenerator createDomain(String containerPath, DomainProps d
 
         return def.getTestDataGenerator(containerPath);
     }
+
+    static public CommandResponse deleteDomain(final String containerPath, final String schema, final String domainName)
+            throws IOException, CommandException
+    {
+        Connection connection = WebTestHelper.getRemoteApiConnection();
+        DropDomainCommand delCmd = new DropDomainCommand(schema, domainName);
+        return delCmd.execute(connection, containerPath);
+    }
+
+    static public boolean doesDomainExists(final String containerPath, final String schema, final String domainName)
+            throws IOException, CommandException
+    {
+        GetQueriesCommand qcmd = new GetQueriesCommand(schema);
+        GetQueriesResponse resp = qcmd.execute(WebTestHelper.getRemoteApiConnection(), containerPath);
+
+        return resp.getQueryNames().contains(domainName);
+    }
```

Comment:
`GetDomainCommand` would be more appropriate here. This could return true for a query that doesn't actually have a domain.
You'd need to catch `CommandException` and check `getStatusCode()` for a 404 for your `false` case, otherwise return true.

Reply:
Ok, I think I updated it the way you suggested. At least the test passed with the change.